### PR TITLE
fixes #24

### LINKED ISSUE #24: NSX Load Balancer pool to support VC-Objects
Currently New-NsxLoadBalancerMemberSpec is used to define a member. Only uses IP address.

Need to modify it to support VC objects like Datacenter, Cluster, Security Group.

### PR DESCRIPTION
**Pull request by nmbradford**
Add support for grouping object support in load balancer pool members

### CODE REVIEW

**Comment by alagoutte** [comment]
My patch is more simple...

Add directly the the "GroupingObject"

		if ($groupingObject) {
		    Add-XmlElement -xmlRoot $xmlMember -xmlElementName "groupingObjectId" -xmlElementText $groupingObject.ExtensionData.MoRef.Value
			Add-XmlElement -xmlRoot $xmlMember -xmlElementName "groupingObjectName" -xmlElementText $groupingObject.name
		}

I known, it is possible to add all type of "object from vCenter" but only VM work on Load Balancer ?


Also don't forget to add Test ;-),

File: module/PowerNSX.psm1 [hunk 1]
```diff
@@ -27192,14 +27192,19 @@ function New-NsxLoadBalancerMemberSpec {
 
     #>
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: Update example ?

File: module/PowerNSX.psm1 [hunk 2]
```diff
@@ -27651,7 +27693,43 @@ function Add-NsxLoadBalancerPoolMember {
...
+            elseif (( $Member -is [VMware.VimAutomation.ViCore.Interop.V1.VIObjectInterop]) -and ( $NsxMemberTypes -contains $Member.ExtensionData.MoRef.Type)) {
+                $MemberMoref = $Member.ExtensionData.MoRef.Value
+            }
+            else {
+                throw "Invalid member specified $($Member)"
+            }
+
+            #Create a new member node
+            Add-XmlElement -xmlRoot $xmlMember -xmlElementName "groupingObjectId" -xmlElementText $MemberMoref
+
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: Same issue

File: module/PowerNSX.psm1 [hunk 3]
```diff
@@ -27225,7 +27230,41 @@ function New-NsxLoadBalancerMemberSpec {
...
+            }
+            elseif (( $Member -is [VMware.VimAutomation.ViCore.Interop.V1.VIObjectInterop]) -and ( $NsxMemberTypes -contains $Member.ExtensionData.MoRef.Type)) {
+                $MemberMoref = $Member.ExtensionData.MoRef.Value
+            }
+            else {
+                throw "Invalid member specified $($Member)"
+            }
+
+            #Create a new member node
+            Add-XmlElement -xmlRoot $xmlMember -xmlElementName "groupingObjectId" -xmlElementText $MemberMoref
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: Add also a groupingObjectName because when you try to edit with GUI get some strange issue (display no name...)

**Comment by nmbradford** [comment]
Thanks for input.

1) VC NSX object is supported (more than just VM object.) so Ill keep my pr :)
2) Thanks for tip on name - ill add before merging.
3) Tests pass, but again, I have one test env that is continually failing that is the cause of the failed test.

**Comment by nmbradford** [comment]
jenkins test this please

**Comment by nmbradford** [comment]
tests succeeded

**Comment by alagoutte** [comment]
>VC NSX object is supported (more than just VM object.) 
But working ? because it is very strange to add a Cluster to LoadBalancer...

I will try

**Pull request merged**